### PR TITLE
Auth: guard against user being removed

### PR DESCRIPTION
**Pull request by maximlt**
A user reported sometimes getting a `KeyError` at line 1008 in `auth.py`.

### CODE REVIEW

File: panel/auth.py [hunk 1]
```diff
@@ -1005,7 +1005,7 @@ async def get_user(handler):
             now_ts = dt.datetime.now(dt.timezone.utc).timestamp()
             expiry = None
             if user in state._oauth_user_overrides:
-                while not state._oauth_user_overrides[user]:
+                while not state._oauth_user_overrides.get(user):
```
Comment (philippjfr) [inline panel/auth.py:1008]: This change isn't safe. If the user override is deleted then the next line after the while loop will error out.

Comment (philippjfr) [reply panel/auth.py:1008]: Sorry, as you rightly pointed out the problem here is that this change can cause this while loop to run forever.

Comment (philippjfr) [reply panel/auth.py:1008]: The correct fix is to ensure that we do not delete the user override while it is refreshing:

```python
if state._oauth_user_overrides.get(user) is not None:
    del state._oauth_user_overrides[user]
```

Comment (maximlt) [reply panel/auth.py]: Alright it makes sense to me! I reverted the fix changes and applied this one.

File: panel/auth.py [hunk 2]
```diff
@@ -1108,7 +1108,9 @@ def _remove_user(self, session_context):
         state._active_users[user] -= 1
         if not state._active_users[user]:
             del state._active_users[user]
-            if user in state._oauth_user_overrides:
+            # Don't remove the user when it's set to an empty dict in the
+            # overrides, as it means it is being refreshed.
```
Comment (philippjfr) [inline panel/auth.py:1112]:
```suggestion
            # Don't remove the user override when it is set to None or
            # is missing, as this means it is being refreshed.
```

**Pull request merged**